File: 1507.reformat-date.js
/*
 * @lc app=leetcode id=1507 lang=javascript
 *
 * [1507] Reformat Date
 */

// @lc code=start
/**
 * @param {string} date
 * @return {string}
 */
const month_to_num = {
    "Jan": "01",
    "Feb": "02",
    "Mar": "03",
    "Apr": "04",
    "May": "05",
    "Jun": "06",
    "Jul": "07",
    "Aug": "08",
    "Sep": "09",
    "Oct": "10",
    "Nov": "11",
    "Dec": "12"
};
var reformatDate = function (date) {
    let data = date.split(' ');
    data[0] = data[0].slice(0, data[0].length - 2);
    if (data[0].length == 1) {
        data[0] = '0' + data[0];
    }
    data[1] = month_to_num[data[1]];
    return data.reverse().join('-');
};
// @lc code=end

File: 1507.reformat-date.cpp
/*
 * @lc app=leetcode id=1507 lang=cpp
 *
 * [1507] Reformat Date
 */
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
    unordered_map<string, int> month_to_num = unordered_map<string, int>(
        {{"Jan", 1}, {"Feb", 2}, {"Mar", 3}, {"Apr", 4}, {"May", 5}, {"Jun", 6}, {"Jul", 7}, {"Aug", 8}, {"Sep", 9}, {"Oct", 10}, {"Nov", 11}, {"Dec", 12}});

public:
    string reformatDate(string date)
    {
        int d = 0;

        for (auto c : date)
        {
            if (isdigit(c))
            {
                d *= 10;
                d += c - '0';
            }
            else
            {
                break;
            }
        }
        string day = '0' + to_string(d);
        string month = '0' + to_string(month_to_num[date.substr(date.find_first_of(' ') + 1, 3)]);

        return date.substr(date.size() - 4) + '-' + month.substr(month.size() - 2) + '-' + day.substr(day.size() - 2);
    }
};
// @lc code=end
